pkg/nodeconfig/payload/payload.go

```diff
@@ -165,8 +195,13 @@ $cni_template=@'
 }
 '@
 
+$hns_network = Retry-GetHnsNetwork
+# If the HNS network is never found, quit.
+if ($null -eq $hns_network) {
+  throw "HNS_NETWORK not found after retries."
```

Comment:
```suggestion
  throw "cannot find HNS_NETWORK with name $hns_network"
```

The error should be a concluding statement and provide supporting information for debugging.

Reply:
if I use $hns_network there, it would always be null. Maybe return the error from the function?

Reply:
>  error from the function

+1

Reply:
WRT to the message content, we can use:

```
cannot find HNS network with name OVNKubernetesHNSNetwork
```

Reply:
This is incorrect. Please revert. `OVNKubernetesHNSNetwork` should not be hardcoded.

The original value `throw "HNS_NETWORK not found after retries."`
Evaluates to `OVNKubernetesHNSNetwork not found after retries."`

This message is fine.